app/services/update_offer_conditions.rb

```diff
@@ -0,0 +1,28 @@
+class UpdateOfferConditions
+  def initialize(application_choice:, conditions:)
+    @application_choice = application_choice
+    @conditions = conditions
+  end
+
+  def call
+    offer = Offer.find_or_initialize_by(application_choice: @application_choice)
+    OfferCondition.where(offer: offer).delete_all
+    @conditions.each do |condition|
```

Comment:
quite a few queries happening here, is there something we can do with bulk `insert` methods?

Reply:
`insert_all!` does let us do this, but we need to provide timestamps as it doesn't by default. It seems nicer to do this in one query though. Adding it now